core/src/main/java/org/apache/iceberg/avro/AvroFileAppender.java

```diff
@@ -113,6 +114,8 @@ private static <D> DataFileWriter<D> newAvroWriter(
       writer.setMeta(entry.getKey(), entry.getValue());
     }
 
+    writer.setEncoder(b -> EncoderFactory.get().blockingDirectBinaryEncoder(b, null));
```

Comment:
Should we also do it https://github.com/apache/iceberg/blob/f4b36a5e553cc8a14208890da2a3adc91ceefbda/core/src/main/java/org/apache/iceberg/data/avro/IcebergEncoder.java#L101 and https://github.com/apache/iceberg/blob/f4b36a5e553cc8a14208890da2a3adc91ceefbda/core/src/main/java/org/apache/iceberg/avro/AvroEncoderUtil.java#L57